### DISCUSSION TITLE
what's the meaning of [SYMM_LABELS, MAP_LABELS]

### DISCUSSION THREAD

**Discussion by Zanejins** [explosion/spaCy#12420] [Help: Coding & Implementations] [answered]
I have used the prodigy tool to mark the entity relationship, and got a dataset in json format. After comparison, my dataset is consistent with the demo file rel_component/assets/annotations.jsonl. Now I want to generate the corresponding train.spacy and other files. I found that parser_data.py is provided to achieve this function, but I encountered this error after execution.

![image](https://user-images.githubusercontent.com/98217021/225001880-634855d3-625d-4c23-9a07-82ff1a8b0e02.png)

After looking at the source code of parser_data, I guess  YMM_LABELS, MAP_LABELS is the problem. How to modify these two maps. The categories of relations I annotated are share-phon, involve and req-rel. Thanks

**Comment by svlandeg** [comment, marked as answer]
Hi!

The `parser_data.py` file was not originally implemented as a generic script that will convert any type of relation annotation - it was developed to support the tutorial and specific data format from https://github.com/explosion/projects/tree/v3/tutorials/rel_component (only). 

If you want to reuse this script, you should really use it as an example on how this type of conversion can be done, but go through all the parts of the code and see what is relevant and what should be changed for your specific use-case.

Alternatively, we recently created this more general script: https://github.com/explosion/projects/blob/v3/tutorials/rel_component/scripts/parse_data_generic.py. Here, you only have to go through the 3 TODO's at the top of the script. Most importantly, `SYMM_LABELS` and `DIRECTED_LABELS` should be lists of labels that occur in your REL annotations and that are symmetrical (A->B implies B->A) or directed (A->B does not necessarily imply B->A) respectively. 

Let us know how you go with this!
